### PR TITLE
Update cross-chain.md

### PR DESCRIPTION
**Pull request by jjyr**

### CODE REVIEW

File: docs/cross-chain.md [hunk 1]
```diff
@@ -2,46 +2,171 @@
 sidebar_position: 6
 ---
 
-# Bridgeless Leap
+# Bridgeless leap
 
-UTXO Stack enables seamless cross-chain interoperability by leveraging the RGB++ protocol[^1]'s isomorphic binding technology `assets leap`. This innovative approach eliminates the need for traditional cross-chain bridges, which can be centralized points of failure.
+UTXO Stack enables seamless cross-chain interoperability between RGB++ layer and branch chains, leveraging the isomorphism architecture. This innovative approach eliminates the need for traditional cross-chain bridges, which are often centralized points of failure, By doing so, it improve the efficiency of asset transfer, allowing tokens to move across branch chains without the need for extended locking periods.
```
Comment (Flouse) [inline docs/cross-chain.md:7]:
```suggestion
UTXO Stack enables seamless cross-chain interoperability between RGB++ layer and branch chains, leveraging an isomorphic architecture. This innovative approach eliminates the need for traditional cross-chain bridges, which are often centralized points of failure. By doing so, it improves the efficiency of asset transfer, allowing tokens to move across branch chains without the need for extended locking periods.
```

File: docs/cross-chain.md [hunk 2]
```diff
@@ -2,46 +2,171 @@
...
 
-The key to UTXO Stack's cross-chain capabilities lies in the way it represents asset ownership. Every RGB++ asset/cell can be bound to a unique UTXO, even the UTXO is on another chain.
+## Branch chain token manager
 
-## Burn and Mint
+The Branch Chain Token Manager, referred to as the token manager, is a system contract embedded within every branch chain. It functions as the issuer of branch chain tokens. When assets are transferred from RGB++ to a branch chain, the original assets are locked on RGB++, and the token manager issue a wrapped token on the branch chain. Conversely, when assets move from a branch chain back to RGB++, the wrapped tokens are burned on the branch chain, and the original assets on RGB++ are unlocked.
```
Comment (Flouse) [inline docs/cross-chain.md:11]:
```suggestion
The Branch Chain Token Manager, referred to as the token manager, is a system contract embedded within every branch chain. It functions as the issuer of branch chain tokens. When assets are transferred from RGB++ layer to a branch chain, the original assets are locked on RGB++ layer, and the token manager issues a wrapped token on the branch chain. Conversely, when assets move from a branch chain back to RGB++ layer, the wrapped tokens are burned on the branch chain, and the original assets on RGB++ layer are unlocked.
```

File: docs/cross-chain.md [hunk 3]
```diff
@@ -2,46 +2,171 @@
...
-CSVBurn is executed on the source chain, and the uUDT contract verifies:
+Assets on the branch chain are represented by [xUDT][xUDT] tokens, with the token manager contract acting as the issuer.
 
-1. At least `X` amount of tokens have been burned in this transaction.   
-2. `src_chain_id` is the chain_id of the current chain.
-3. `dst_chain_id` and `out_point` refer to the `out_point` on another chain.
+## Cross chain message queue
 
-### `CSVMint(X, burn_tx)`
+On RGB++:
```
Comment (Flouse) [inline docs/cross-chain.md:21]:
```suggestion
On RGB++ layer:
```

File: docs/cross-chain.md [hunk 4]
```diff
@@ -2,46 +2,171 @@
...
+The cross-chain message queue is implemented as user contracts. This contract includes a branch chain light client, which tracks the branch chain's state, and a queue contract that maintains messages sent to the branch chain.
 
-CSVMint is executed on the destination chain, and the uUDT contract verifies:
+On Branch Chains:
+The message queue is implemented as a system contract. This setup allows branch validators to read messages from RGB++ and submit cross-chain messages to the system contract. Branch chain users can also send messages to the system contract, and validators act as relayers by submitting these messages to RGB++.
```
Comment (Flouse) [inline docs/cross-chain.md:25]:
```suggestion
The message queue is implemented as a system contract. This setup allows branch validators to read messages from RGB++ and submit cross-chain messages to the system contract. Branch chain users can also send messages to the system contract, and validators act as relayers by submitting these messages to RGB++ layer.
```

File: docs/cross-chain.md [hunk 5]
```diff
@@ -2,46 +2,171 @@
...
+Assets on the branch chain are represented by [xUDT][xUDT] tokens, with the token manager contract acting as the issuer.
 
-1. At least `X` amount of tokens have been burned in this transaction.   
-2. `src_chain_id` is the chain_id of the current chain.
-3. `dst_chain_id` and `out_point` refer to the `out_point` on another chain.
+## Cross chain message queue
 
-### `CSVMint(X, burn_tx)`
+On RGB++:
+The cross-chain message queue is implemented as user contracts. This contract includes a branch chain light client, which tracks the branch chain's state, and a queue contract that maintains messages sent to the branch chain.
```
Comment (Flouse) [inline docs/cross-chain.md:22]:
```suggestion
The cross-chain message queue is implemented as user contracts. This contract includes a branch chain light client, which tracks the branch chain's state, and a queue contract that maintains messages related to the branch chain.
```

File: docs/cross-chain.md [hunk 6]
```diff
@@ -2,46 +2,171 @@
...
-3. Generate a CSV proof on the source chain.
-4. Call `CSVMint` on the destination chain and provide the `CSV proof`.
-5. The `mint out_point` must be used as one of the mint transaction's inputs, and X amount of tokens will be minted.
+Batch Request Lock Fields
+
+- **cancel_timeout**: Specifies the number of blocks after which the request can be canceled.
+- **owner_lock_hash**: Allows the request to be canceled with this lock hash.
+- **request**: Serialized byte structure containing the request details.
+
+By creating request cell, users permit aggregators to collect and batch these requests. This approach allows multiple users to interact with the message queue simultaneously and efficiently.
```
Comment (Flouse) [inline docs/cross-chain.md:39]:
```suggestion
By creating a request cell, users permit aggregators to collect and batch these requests. This approach allows multiple users to interact with the message queue simultaneously and efficiently.
```

File: docs/cross-chain.md [hunk 7]
```diff
@@ -2,46 +2,171 @@
...
+  owner_lock_hash: Byte32,
+  amount: u128,
+  // Asset type
+  asset_type: Byte32,
+}
+```
+
+## Leap
+
+We use UTXO token as an example to describe the life cycle of cross chain assets.
```
Comment (Flouse) [inline docs/cross-chain.md:131]: suggestion:
UTXO token -> $UTXO token

File: docs/cross-chain.md [hunk 8]
```diff
@@ -2,46 +2,171 @@
...
+
+### RGB++ to Branch chain
```
Comment (Flouse) [inline docs/cross-chain.md:133]:
```suggestion
### RGB++ layer to Branch chain
```
IMO, RGB++ is a protocol by default.
So I think it's better to name CKB as RGB++ layer.

File: docs/cross-chain.md [hunk 9]
```diff
@@ -2,46 +2,171 @@
...
+| Step | CKB | Branch | CKB Message Queue | Branch Message Queue | Description |
+|------|-----|--------|-----------|-------------|-------------|
+| 1 | UTXO token cell(lock=user, type=UTXO token) | - | - | - | User's assets on CKB |
+| 2 | Request cell(lock=request_lock, type=UTXO token) | - | - | - | Initiates cross-chain transfer request on CKB |
+| 3 | Custodian cell(lock=custodian_lock, type = UTXO token) | - | outbox: CKBToBranchMessage | - | Lock assets on CKB, enqueue CKBToBranchMesasge |
+| 4 | - | - | outbox: CKBToBranchMessage | inbox: CKBToBranchMessage | submit message to Branch |
+| 5 | - | Wrapped UTXO token cell(lock=user, type = WUTXO token) | outbox: CKBToBranchMessage | - | Mints assets on Branch |
+| 6 | - | - | - | - | Acknowledges and remove processed request from CKB outbox |
+
+### Branch chain to RGB++
```
Comment (Flouse) [inline docs/cross-chain.md:144]:
```suggestion
### Branch chain to RGB++ layer
```

File: docs/cross-chain.md [hunk 10]
```diff
@@ -2,46 +2,171 @@
...
+  outbox: Vec<CKBMessage>,
+  // Receive queue
+  inbox: Vec<BranchMessage>,
+}
+```
+
+Key operations include:
+
+* PushRequest: Submit cross chain transfer message.
+* Ack: Confirms messages have been processed by providing branch chain block headers and proofs.
```
Comment (chesterchen2010) [inline docs/cross-chain.md:59]:
```suggestion
* Ack: Confirm messages have been processed by providing branch chain block headers and proofs.
```

File: docs/cross-chain.md [hunk 11]
```diff
@@ -2,46 +2,171 @@
...
+* HandleRecv: Handles the receive queue.
```
Comment (chesterchen2010) [inline docs/cross-chain.md:60]:
```suggestion
* HandleRecv: Handle the receive queue.
```

File: docs/cross-chain.md [hunk 12]
```diff
@@ -2,46 +2,171 @@
...
+  outbox: Vec<Bytes(BranchMessage)>,
+  // Receive queue
+  inbox: Vec<CKBMessage>,
+}
+```
+
+Key operations include:
+
+* PushRequest: Submit cross chain transfer message.
+* HandleRecv: Handles the receive queue.
```
Comment (chesterchen2010) [inline docs/cross-chain.md:80]:
```suggestion
* HandleRecv: Handle the receive queue.
```

File: docs/cross-chain.md [hunk 13]
```diff
@@ -2,46 +2,171 @@
...
+Assets on the branch chain are represented by [xUDT][xUDT] tokens, with the token manager contract acting as the issuer.
 
-1. At least `X` amount of tokens have been burned in this transaction.   
-2. `src_chain_id` is the chain_id of the current chain.
-3. `dst_chain_id` and `out_point` refer to the `out_point` on another chain.
+## Cross chain message queue
 
-### `CSVMint(X, burn_tx)`
+On RGB++ layer:
+The cross-chain message queue is implemented as user contracts. This contract includes a branch chain light client, which tracks the branch chain's state, and a queue contract that maintains messages related to the branch chain.
```
Comment (Flouse) [inline docs/cross-chain.md:22]:
```suggestion
The cross-chain message queue is implemented as application contracts. This contract includes a branch chain light client, which tracks the branch chain's state, and a queue contract that maintains messages related to the branch chain.
```

**Review by Flouse — APPROVED** [review]

**Review by EthanYuan — APPROVED** [review]

**Pull request merged**